addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -541,6 +605,14 @@ def _get_availabilities(self, product, quantity, product_info, bom_key, quantiti
 
     @api.model
     def _get_stock_availability(self, product, quantity, product_info, quantities_info):
+        closest_forecasted_values = self._context.get('components_closest_forecasted', {}).get(product.id)
+        closest_forecasted = None
+        if closest_forecasted_values:
+            closest_forecasted = closest_forecasted_values.pop(0)
```

Comment:
Sorry I still have a issue, in case of a single component being used multiple times in the same BoM.
In that case a single replenishment can only be used once (since it's `pop`'ed), regardless of its quantity.
See the following case, for both I have an ongoing PO containing the required 6 qty.
v16 :
![image](https://github.com/odoo/odoo/assets/18055894/a28bc1ff-c037-486c-b7b7-a45ea453e05c)
This PR :
![image](https://github.com/odoo/odoo/assets/18055894/7200770f-bced-4795-b5c7-14dd6c9d8d06)

Reply:
Yeah I know. I pushed that a bit too quickly sorry. I'm still rewriting it locally because I missed some edge cases. At this point this PR is giving me headaches but I'll see if I find a way to make it work

Reply:
No worries, I didn't see changes in a while on the PR so I thought I'd have a look.
I know this report is a pain, thanks a lot for looking at it. Just ping me when you want me to have a look then! :slightly_smiling_face: